app_merge.go

```diff
@@ -58,12 +58,12 @@ func AssembleHandler(router *mux.Router, apps ...*APIApp) (http.Handler, error)
 
 // MergeApplications takes a number of gimlet applications and
 // resolves them, returning an http.Handler.
-func MergeApplications(apps ...*APIApp) (http.Handler, error) {
+func MergeApplications(router *mux.Router, apps ...*APIApp) (http.Handler, error) {
 	if len(apps) == 0 {
 		return nil, errors.New("must specify at least one application")
 	}
 
-	return AssembleHandler(mux.NewRouter().UseEncodedPath(), apps...)
+	return AssembleHandler(router, apps...)
```

Comment:
Should this still call `UseEncodedPath()` on the router to preserver behavior? Or maybe create a new router if that passed in router is `nil`?
```
if router == nil {
        router = mux.NewRouter().UseEncodedPath()
}
```

Reply:
Another option would be to keep this function as is and create a new function that does the same thing but also accepts a custom router:
```
func MergeApplicationsWithRouter(...)
```

Reply:
done! (the second route)

Reply:
oh, I didn't see that last suggestion. Let me do that.

Reply:
okay. really done now.